osu.Game/Screens/Multi/RoomManager.cs

```diff
@@ -142,6 +143,8 @@ public void PartRoom()
             joinedRoom = null;
         }
 
+        private readonly List<int> roomsFailedUpdate = new List<int>();
```

Comment:
How about making this a

```suggestion
        private readonly HashSet<int> roomsFailedUpdate = new HashSet<int>();
```

and just skipping the `Contains()` check in the usage site entirely?

Reply:
@bdach i've refactored this slightly to avoid throwing exceptions each poll (i think this is saner). please double-check.